backtracking/subset_sum.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file
+ * @brief Implementation of the [Subset
+ * Sum](https://en.wikipedia.org/wiki/Subset_sum_problem) problem.
+ * @details
+ * We are given an array and a sum value. The algorithms find all
+ * the subsets of that array with sum equal to the given sum and return such
+ * subsets count. This approach will have exponential time complexity.
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief Backtracking algorithms
+ */
+namespace backtracking {
+/**
+ * @namespace Subsets
+ * @brief Functions for the [Subset
+ * Sum](https://en.wikipedia.org/wiki/Subset_sum_problem) problem.
+ */
+namespace subset_sum {
+/**
+ * @brief The main function implements count of subsets
+ * @param sum is the required sum of any subset
+ * @param in_arr is the input array
+ * @returns count of the number of subsets with required sum
+ */
+
+uint64_t subset_sum(int32_t sum, const std::vector<int32_t> &in_arr) {
```

Comment:
namespace is similar to function name. Function is returning count of subset. You can think of some other name.